scripts/docs_config/glossary.rst

```diff
@@ -0,0 +1,100 @@
+Glossary
+==========================================================
+
+Homogeneous Memory
+  A set of memory used in a system that is composed of a single memory type of 
```

Comment:
It's still a bit convoluted, can't it just be: 

```
Homogeneous Memory System

A system that operates on a single type of memory implemented using a single technology.
```

```
Heterogenous Memory System

A system that operates on multiple types of memories, possibly implemented using different technologies, often managed by different entities.
```

Or something like this?